c7n/policy.py

```diff
@@ -700,12 +701,35 @@ def is_lambda(self):
         return True
 
     def validate(self):
+        self._check_deprecation('resource', self.resource_manager)
```

Comment:
i'd just pass the resource type here so you can get consistent message generation below.